### ISSUE TITLE
Clang should not warn about exact-width specifiers for printf

### ISSUE THREAD

**Issue by sstwcw** [llvm/llvm-project#76377]
Compiling this code gives a warning.  I expect no warning.  GCC does not give a warning since version 13.  It looks like the front end does not recognize that specifiers for exact-width types are allowed.

```C
#include <stdint.h>
#include <stdio.h>

int main(void) {
  printf("%w64d\n", INT64_C(0));
  return 0;
}
```

```
clang -Wall -std=c2x print.c
print.c:5:13: warning: invalid conversion specifier '6' [-Wformat-invalid-specifier]
    5 |   printf("%w64d\n", INT64_C(0));
      |           ~~^
```

My version is 18.0.0git 6452395561ea.

**Comment by hazohelet** [comment]
Looks relevant to: https://github.com/llvm/llvm-project/issues/71740

**Comment by AaronBallman** [comment]
Closing as a duplicate of #71740.
